fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -817,6 +818,48 @@ class BrowserRobot {
         assertItemWithResIdExists(itemWithResId("errorTryAgain"))
     }
 
+    fun verifyOpenLinksInAppsCFRExists(exists: Boolean) {
+        for (i in 1..RETRY_COUNT) {
+            try {
+                assertItemWithResIdExists(
+                    itemWithResId("$packageName:id/banner_container"),
+                    exists = exists,
+                )
+                assertItemWithResIdAndTextExists(
+                    itemWithResIdContainingText(
+                        "$packageName:id/banner_info_message",
+                        getStringResource(R.string.open_in_app_cfr_info_message_2),
+                    ),
+                    itemWithResIdContainingText(
+                        "$packageName:id/dismiss",
+                        getStringResource(R.string.open_in_app_cfr_negative_button_text),
+                    ),
+                    itemWithResIdContainingText(
+                        "$packageName:id/action",
+                        getStringResource(R.string.open_in_app_cfr_positive_button_text),
+                    ),
+                    exists = exists,
+                )
+            } catch (e: AssertionError) {
+                if (i == RETRY_COUNT) {
+                    throw e
+                } else {
+                    browserScreen {
+                    }.openThreeDotMenu {
+                    }.refreshPage {
+                        progressBar.waitUntilGone(waitingTimeLong)
+                    }
+                }
+            }
+        }
+    }
+
+    fun clickOpenLinksInAppsDismissCFRButton() =
+        itemWithResIdContainingText(
+            "$packageName:id/dismiss",
+            getStringResource(R.string.open_in_app_cfr_negative_button_text),
+        ).click()
```

Comment:
could this be replaced by clickPageObject()?

Reply:
I would want to leave it as it is.

1. It's not a page object
2. I don't see the benefit of using it